src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultStates = [];
+
+  let stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(stateCopy, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        stateCopy = {};
+        break;
+      }
+
+      default:
+        continue;
```

Comment:
The switch statement is used correctly here to handle different action types. However, the default case should not use 'continue' as it's not inside a loop that can be continued. It would be better to throw an error or handle an unexpected action type more gracefully.